src/main/groovy/edu/oregonstate/mist/osuevents/resources/EventsResource.groovy

```diff
@@ -0,0 +1,239 @@
+    @Path('{id: [0-9a-zA-Z-]+}')
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getByID(@Auth AuthenticatedUser _,
+                            @PathParam('id') String id) {
+        ResourceObject event = eventsDAO.getById(id)
+        ok(getResultObject(event)).build()
+    }
+
+/**
+ * GET all events
+ */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getEvents(@Auth AuthenticatedUser _,
+                              @QueryParam('format') String format) {
+        def events = eventsDAO.getEvents()
+//@TODO add methods for returning csv and ics formats
+//        else if (format == "csv") {
+//
+//        } else if (format == "ics") {
+//
+//        } else {
+//            return badRequest("Invalid format value. Valid formats are csv or ics.").build()
+//        }
+        ok(getResultObject(events)).build()
+    }
+/**
+ * POST an event
+ */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createEvent(@Auth AuthenticatedUser _,
+                                @Valid ResultObject newResultObject) {
+        ResourceObject newResourceObject
+        Event newEvent
+
+        try {
+            newResourceObject = newResultObject.data
+            newEvent = newResultObject.data.attributes
+        } catch (GroovyCastException e) {
+            return badRequest(ErrorMessages.unknownFields).build()
+        } catch (Exception e) {
+            logger.error("Exception while calling createEvent", e)
+            return internalServerError(ErrorMessages.unexpectedException).build()
+        }
+
+        //generate id if one wasn't included in the request
+        newResourceObject.id = newResourceObject.id ?: randomUUID() as String
+
+        if (!newResourceObject.id.matches(uuidRegEx) || eventsDAO.getById(newResourceObject.id)) {
+            return conflict().build()
+        }
+
+        //get custom fields and filters ready for DAO
+        String customFieldData = JsonOutput.toJson(newEvent.customFields)
+        String filterData = JsonOutput.toJson(newEvent.filters)
+
+        try {
+            newEvent.instances.each {
+                eventsDAO.createInstance(
+                        it.id,
+                        newResourceObject.id,
+                        InstanceMapper.formatForDB(it.start.toString()),
+                        InstanceMapper.formatForDB(it.end.toString())
+                )
+            }
+        } catch (DateTimeParseException e) {
+            return badRequest(ErrorMessages.parseDate).build()
+        } catch (MissingMethodException e) {
+            return badRequest(ErrorMessages.processInstance).build()
+        } catch (Exception e) {
+            logger.error("Exception while calling createEvent", e)
+            return internalServerError(ErrorMessages.unexpectedException).build()
+        }
+
+        eventsDAO.createEvent(newResourceObject.id, newEvent, filterData, customFieldData)
+
```

Comment:
Quite a bit of what this method does is validation. Can the validation be moved to another method that returns a list of errors? This method could then generate an array of errors to return to the user.